framework/doc/content/source/meshgenerators/MeshGenerator.md

```diff
@@ -79,15 +41,102 @@ For instance, take a look at the following input file:
   []
 []
 
-[Mesh]
-  type = MeshGeneratorMesh
-[]
-
 [Outputs]
   exodus = true
 []
 ```
 
-Structurally, notice that one can have several `MeshGenerator` blocks, but only one `Mesh` block of type `MeshGeneratorMesh`.
+Let's analyze the different steps. First, we use a [GeneratedMeshGenerator.md] to create a 3D mesh labeled `gmg` (here it's simply a regular cube). After that, we use this mesh as an input (via `input = gmg`) to create a larger mesh using a [TiledMeshGenerator.md].
+
+## Developing a MeshGenerator
+
+For mesh generators that create a mesh, care must be taken as to the parallel type of the mesh. You will override the `generate()` method, obtain a base mesh object, and then return your generated mesh. For obtaining the base mesh object, there are three options:
+
+- If the mesh to be generated is agnostic to the parallel type (that is, it can be either replicated or distributed), use `MeshGenerator::buildMeshBaseObject()` to generate the base mesh. See [GeneratedMeshGenerator.md] for an example.
+- If the mesh to be generated is only replicated, use `MeshGenerator::buildReplicatedMesh()` to generate the base mesh.
+- If the mesh to be generated is only distributed, use `MeshGenerator::buildDistributedMesh()` to generate the base mesh. See [DistributedRectilinearMeshGenerator.md] for an example.
+
+You should +only+ use these APIs to create the base mesh in generators that create meshes. This is very important, because the internal preparation of the meshes for use is dependent on the parallel type of the mesh.
+
+For mesh generators that modify an existing mesh, you should have as an input parameter a `MeshGeneratorName` (or multiple, as a `std::vector<MeshGeneratorName>` if applicable) to obtain the mesh(es) to modify. You can then obtain said meshes via `MeshGenerator::getMesh()` and `MeshGenerator::getMeshByName()`. For examples, see [RenameBoundaryGenerator.md] and [StitchedMeshGenerator.md]. You then act on said meshes by overriding the `generate()` method, and returning the resulting mesh.
+
+## Showing Mesh Information
+
+The parameter [!param](/Mesh/GeneratedMeshGenerator/show_info) prints to screen detailed information about a mesh immediately after it is generated. It can be used to quickly verify properties about the mesh such as: sidesets, subdomains, and volumes.
+
+Take the simple example:
+
+```
+[Mesh]
+  [gmg]
+    type = GeneratedMeshGenerator
+    dim = 2
+    nx = 2
+    ny = 2
+    show_info = true
+  []
+```
+
+The above will result in on-screen output similar to the following (ran with two processes):
 
-Let's analyze the different steps. First, we use a `GeneratedMeshGenerator` to create a 3D mesh labeled `gmg` (here it's simply a regular cube). After that, we use this mesh as an input (via `input = gmg`) to create a larger mesh using a `TiledMeshGenerator`.
+```
+GeneratedMeshGenerator 'gmg':  Mesh Information:
```

Comment:
After looking at this, I am not sure I like the prefix. Can you just put the type and name at the beginning of the mesh info dump. Either has a heading or in the information list (or both)

```
\\ MyGenerator::foo \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
Mesh Information:
   Type: MyGenerator
   Name: foo
...
```

Reply:
I thought about this too, but when you have a ton of MGs that print output it's really nice to be able to search for them easily and see when one ends and one begins by just scrolling and seeing the difference in width on the left.